probe-rs/src/probe.rs

```diff
@@ -722,6 +722,11 @@ impl DebugProbeInfo {
     pub fn open(&self, lister: &Lister) -> Result<Probe, DebugProbeError> {
         lister.open(DebugProbeSelector::from(self))
     }
+
+    /// Returns whether this info was returned by a particular probe factory.
+    pub fn is_probe_type<F: ProbeFactory>(&self, factory: &F) -> bool {
```

Comment:
This naming still sounds weird to me, if there is no `DebugProbeType` anymore then we should rename the `probe_type` field, and call this function something else.

Reply:
I'm open to suggestions